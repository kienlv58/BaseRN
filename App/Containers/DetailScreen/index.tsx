import React, { ReactElement } from 'react';
import { Text, View } from 'react-native';
import ButtonIcon from 'App/Components/ButtonIcon';
import { BaseScreenProps } from 'App/@types/screen-type';

const DetailScreen = (props: BaseScreenProps): ReactElement => {
  //TODO
  console.log(props);

  return (
    <View>
      <Text>this is detail screen</Text>
      <ButtonIcon />
    </View>
  );
};
export default DetailScreen;
