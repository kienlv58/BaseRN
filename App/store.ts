import configureStore from 'App/Lib/configureStore';

const store = configureStore();

export default store;
