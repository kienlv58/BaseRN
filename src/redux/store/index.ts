import {configureStore} from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';
import createDebugger from 'redux-flipper';
import rootReducer from '../slices/rootReducer';

const persistConfig = {
  key: 'root',
  version: 1,
  storage: AsyncStorage,
  whitelist: ['global'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware => {
    const middlewares = getDefaultMiddleware({
      thunk: true,
      immutableCheck: false,
      // serializableCheck: false, // shoud enable. Now disable for skip warning log => maybe bugs
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    });
    console.log('__DEV__', __DEV__);
    if (__DEV__) {
      middlewares.push(createDebugger());
    }
    middlewares.push(thunk);
    return middlewares;
  },
  devTools: process.env.NODE_ENV !== 'production',
});
export const persistor = persistStore(store);
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
export default store;
