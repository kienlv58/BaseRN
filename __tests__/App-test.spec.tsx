/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import 'react-native-gesture-handler';
import { Provider } from 'react-redux';

import store from 'App/store';
import DetailScreen from '../App/Containers/DetailScreen';
import { mockBaseScreenProps } from './__mocks__/base-screen.mock';
// Note: test renderer must be required after react-native.

describe('Home Screen', () => {
  it('renders correctly', async () => {
    expect(
      renderer
        .create(
          <Provider store={store}>
            <DetailScreen {...mockBaseScreenProps} />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();

    expect(1).toEqual(1);
  });
});
