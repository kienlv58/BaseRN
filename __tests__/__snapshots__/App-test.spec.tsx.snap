// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<View>
  <Text>
    this is detail screen
  </Text>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 200,
        "opacity": 1,
      }
    }
  >
    <Text>
      test button
    </Text>
  </View>
</View>
`;
